.travis.yml

```diff
@@ -1,32 +1,23 @@
 # Modified from
 # https://github.com/calliope-project/calliope/blob/master/.travis.yml
 
-language: python
+language: bash
 sudo: false # Use container-based infrastructure
 
+
 matrix:
   include:
-    - env:
-      - PYTHON_VERSION="2.7"
+    - os:
+      - windows
+      - linux
     - env:
       - PYTHON_VERSION="3.6"
-    - env:
       - PYTHON_VERSION="3.7"
 
 before_install:
-  - if [[ "$PYTHON_VERSION" == "2.7" ]]; then
-      wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O miniconda.sh;
-    else
-      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
-    fi
-  - bash miniconda.sh -b -p $HOME/miniconda
-  - export PATH="$HOME/miniconda/bin:$PATH"
-  - hash -r
-  - conda config --set always_yes yes --set changeps1 no
-  - conda update -q conda
-  # Useful for debugging any issues with conda
-  - conda info -a
-  - source $HOME/miniconda/etc/profile.d/conda.sh
+  - wget https://raw.githubusercontent.com/trichter/conda4travis/latest/conda4travis.sh -O conda4travis.sh
+  - source conda4travis.sh
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then echo "  - coincbc" >> environment.yaml; fi;
```

Comment:
Very creative!

Reply:
:) best solution for now. Conditional installation in `environment.yaml` is unfortunately not supported, and I did not want to set up a third environment file